_applications/jgrafzahl.md

```diff
@@ -0,0 +1,129 @@
+. A production-ready application would separate the processor from the web UI
+by an operational data store, i.e. a database or in-memory data grid.
+
+For jGraf Zahl to have anything to do she needs some data to
+consume. To help her out here we also provide two other pods.
+
+1. Apache Kafka running entirely in one pod. This is not
+   production-ready because it does nothing to provide replication,
+   durability or scalability of the topic used by jGraf Zahl. What it
+   does provide is just enough Kafka to demonstrate stream processing.
+
+2. A source of some data to count. For this we provide a word
+   fountain, generating words to the topic that jGraf Zahl will
+   consume.
+
+<h1 id="installation">Installation</h1>
+
+Installing and deploying jGraf Zahl utilizes [Oshinko
+S2I](https://github.com/radanalyticsio/oshinko-s2i), specifically the
+[Oshinko java](https://hub.docker.com/r/radanalyticsio/radanalytics-java-spark/)
+builder. [S2I](https://docs.openshift.com/enterprise/latest/architecture/core_concepts/builds_and_image_streams.html#source-build)
+is a technology for taking a source repository that has a specific
+layout and building it into a container image that is then deployed
+as a pod on OpenShift.
+
+The [word fountain](https://github.com/mattf/word-fountain) component
+will also use S2I, but the [default python
+builder](https://docs.openshift.com/enterprise/latest/using_images/s2i_images/python.html)
+provided by OpenShift, because there is no dependency on Apache Spark.
+
+Apache Kafka is more similar to infrastructure for the other
+components and not an application itself, so instead of using S2I, it
+is directly deployed from a template and pre-built container images.
+
+First, make sure you are connected to an OpenShift cluster and are in
+a project with Oshinko installed. See [Get Started](/get-started) if
+you need help.
+
+Second, load the Apache Kafka infrastructure components into your
+project and start them.
+
+```
+oc create -f https://raw.githubusercontent.com/mattf/openshift-kafka/master/resources.yaml
+oc new-app apache-kafka
+```
+
+Third, launch the word fountain, so jGraf Zahl will have something to
+count. The word fountain uses the `SERVERS` environment variable to
+find the Apache Kafka deployment to use. In the second step, when you
+created `apache-kafka` you created a service with the same name on
+port 9092. Note: The first time this step and the next run you'll have
+to wait for the builder images to be pulled down from the internet, so
+if you're on a thin pipe you may want to start both at the same time
+and grab a drink.
+
+```
+oc new-app openshift/python-27-centos7~https://github.com/mattf/word-fountain -e SERVERS=apache-kafka:9092
+```
+
+Fourth, launch jGraf Zahl herself, using the Oshinko java S2I
+builder.
+
+```
+oc new-app --template=oshinko-java-spark-build-dc \
+           -p APPLICATION_NAME=jgrafzahl \
+           -p GIT_URI=https://github.com/elmiko/jgrafzahl \
```

Comment:
in general i would like to see more stuff getting migrated to the radanalytics org, matt has those links to word-fountain and openshift-kafka in the grafzahl tutorial as well. i'm not sure if they will be migrated to radanaytics, but i can certainly move jgrafzahl over.

good spot, and i think we need a stronger policy about this going forward.